diff --git b/kernel/config.c a/kernel/config.c
index cd1d2f6..672a1e3 100644
--- b/kernel/config.c
+++ a/kernel/config.c
@@ -34,7 +34,7 @@ void iet_procfs_exit(void)
 	for (i = 0; i < ARRAY_SIZE(iet_proc_entries); i++)
 		remove_proc_entry(iet_proc_entries[i].name, proc_iet_dir);
 
-	proc_remove(proc_iet_dir);
+	remove_proc_entry(proc_iet_dir->name, proc_iet_dir->parent);
 }
 
 int iet_procfs_init(void)
@@ -46,9 +46,10 @@ int iet_procfs_init(void)
 		goto err;
 
 	for (i = 0; i < ARRAY_SIZE(iet_proc_entries); i++) {
-		ent = proc_create(iet_proc_entries[i].name, 0, proc_iet_dir,
-                          iet_proc_entries[i].fops);
-		if (!ent)
+		ent = create_proc_entry(iet_proc_entries[i].name, 0, proc_iet_dir);
+		if (ent)
+			ent->proc_fops = iet_proc_entries[i].fops;
+		else
 			goto err;
 	}
 
