Index: iscsitarget/kernel/event.c
===================================================================
--- iscsitarget.orig/kernel/event.c	2013-04-12 13:12:09.687313731 +0530
+++ iscsitarget/kernel/event.c	2013-04-12 13:12:52.487312886 +0530
@@ -26,7 +26,7 @@
 		rlen = NLMSG_ALIGN(nlh->nlmsg_len);
 		if (rlen > skb->len)
 			rlen = skb->len;
-		ietd_pid = NETLINK_CB(skb).pid;
+		ietd_pid = NETLINK_CB(skb).portid;
 		WARN_ON(ietd_pid == 0);
 		if (nlh->nlmsg_flags & NLM_F_ACK)
 			netlink_ack(skb, nlh, 0);
@@ -77,7 +77,6 @@
 
 	nl = netlink_kernel_create(&init_net,
 				   NETLINK_IET,
-				   THIS_MODULE,
 				   &cfg);
 	if (!nl)
 		return -ENOMEM;
